src/components/calculator.js

```diff
@@ -1,41 +1,59 @@
 import React from 'react';
 import './calculator.css';
+import calculate from '../logic/calculate';
+import '../logic/operate';
 
 class Calculator extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: 0,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((previousState) => calculate(previousState, e.target.innerText));
   }
 
   render() {
+    let val = 0;
+    const { total, next } = this.state;
+    if (next) {
+      val = next;
+    } else if (total) {
+      val = total;
+    }
+
     return (
       <div className="wrapper">
 
-        <input type="text" className="operator display" id="calc-screen" placeholder="0" disabled />
+        <input type="text" className="operator display" id="calc-screen" placeholder="0" value={val} />
 
-        <button type="button" className="operator all-clear" name="AC">AC</button>
-        <button type="button" className="operator Plus-minus operations" name="+/-">+/-</button>
-        <button type="button" className="operator" name="%">%</button>
-        <button type="button" className="operator operation" name="+">+</button>
+        <button type="button" className="operator all-clear" name="AC" onClick={this.handleClick}>AC</button>
+        <button type="button" className="operator Plus-minus operations" name="+/-" onClick={this.handleClick}>+/-</button>
+        <button type="button" className="operator" name="%" onClick={this.handleClick}>%</button>
+        <button type="button" className="operator operation" name="+" onClick={this.handleClick}>+</button>
 
-        <button type="button" className="operator number" name="7">7</button>
-        <button type="button" className="operator number" name="8">8</button>
-        <button type="button" className="operator number" name="9">9</button>
-        <button type="button" className="operator operation" name="x">&times;</button>
+        <button type="button" className="operator number" name="7" onClick={this.handleClick}>7</button>
+        <button type="button" className="operator number" name="8" onClick={this.handleClick}>8</button>
+        <button type="button" className="operator number" name="9" onClick={this.handleClick}>9</button>
+        <button type="button" className="operator operation" name="x" onClick={this.handleClick}>&times;</button>
```

Comment:
- [ ] When I try to use the multiplication sign and hit the equal to, the app crashes. Kindly take a look at your implementation for the multiplication symbol so that it's possible to perform multiplication in the app. Currently, if you try `2 * 4` and hit the `=` sign, the app crashes. Please fix that 👍🏽